skiko/src/jvmMain/cpp/linux/redrawer.cc

```diff
@@ -0,0 +1,142 @@
+#include <jawt_md.h>
+#include <GL/gl.h>
+#include <GL/glx.h>
+#include <X11/X.h>
+#include <X11/Xlib.h>
+#include <X11/Xresource.h>
+#include <cstdlib>
+#include <unistd.h>
+#include <stdio.h>
+
+typedef GLXContext (*glXCreateContextAttribsARBProc)(Display *, GLXFBConfig, GLXContext, Bool, const int *);
+
+extern "C" jboolean Skiko_GetAWT(JNIEnv *env, JAWT *awt);
+
+extern "C"
+{
+    JNIEXPORT jlong JNICALL Java_org_jetbrains_skiko_redrawer_LinuxRedrawerKt_lockDrawingSurfaceNative(JNIEnv *env, jobject redrawer, jobject layer)
+    {
+        JAWT awt;
+        awt.version = (jint)JAWT_VERSION_9;
+        if (!Skiko_GetAWT(env, &awt))
+        {
+            fprintf(stderr, "JAWT_GetAWT failed! Result is JNI_FALSE\n");
+            return 0;
+        }
+
+        JAWT_DrawingSurface *ds = awt.GetDrawingSurface(env, layer);
+        ds->Lock(ds);
+
+        return static_cast<jlong>(reinterpret_cast<uintptr_t>(ds));
+    }
+
+    JNIEXPORT jlong JNICALL Java_org_jetbrains_skiko_redrawer_LinuxRedrawerKt_unlockDrawingSurfaceNative(JNIEnv *env, jobject redrawer, jlong drawingSurfacePtr)
+    {
+        JAWT_DrawingSurface *ds = reinterpret_cast<JAWT_DrawingSurface *>(static_cast<uintptr_t>(drawingSurfacePtr));
+        JAWT awt;
+        awt.version = (jint)JAWT_VERSION_9;
+        if (!Skiko_GetAWT(env, &awt))
```

Comment:
Why not store AWT pointer and pass it to calls?

Reply:
Skiko_GetAWT already caches the pointer.

We can store AWT pointer inside each Java class, but there will be no performance gain and there still be the same amount of code.

Reply:
But I think we can store global pointer to AWT in Java:

```
object NativeAwt {
  val ptr: Long = getAWT()
}
```
So in this case the code will be more robust. But it is better to do it in a separate PR.